SPECS/python-rich/python-rich.spec

```diff
@@ -10,7 +10,12 @@ Source0:        %{url}/archive/v%{version}/rich-%{version}.tar.gz
 
 BuildArch:      noarch
 
-Patch0:          3229.patch
+Patch0:         3229.patch
+Patch1:         ptest-warning.patch
+# This patch modifies the tests to set the OLD_PYGMENTS version
+# to the current one in Azure Linux, 2.5.2. Once python-pygments
+# version is updated >= 2.14.0, this patch can be removed.
```

Comment:
The patch targets only the 2.5.2 version. If we ever update to 2.6.0 for instance, the test will get re-enabled, while they will still fail, if I understand the comment correctly.

Reply:
That is correct, in this regard I took the existing logic from `tests/test_syntax.py` and added the logic to additional files and modified the checked version for the one present in azure linux 3.0 (the original check only targeted 2.13.0, hence the tests failing for us in the first place).

In the event that we do upgrade to a version of `python-pygments` lower than 2.14.0, this patch would need to be modified in the future.

Reply:
Links to the code in question and commit which changed these tests accordingly:
https://github.com/Textualize/rich/blob/master/tests/test_syntax.py#L29
https://github.com/Textualize/rich/pull/3139/files

Reply:
Since we're interested in pre-2.14.0 versions couldn't we compare if `pygments` is < 2.14.0 now, so we don't have to worry about this until we reach 2.14.0 or higher? This looks like a promising way to [compare versions in Python](https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python) you could try. It may require an `BuildRequires: python3-packaging` but I think that shouldn't be a problem - if it's inside the `0%{?with_check}` section, then there's no risk of a circular dependency as well.

Reply:
Updated to use the Version changes as detailed. Replace the comparison with a '<=' 2.13.0 as that is the targeted version from rich to start. Seems to work as expected.

buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=571115&view=results